swo_aws_extension/flows/steps/setup_context.py

```
@@ -0,0 +1,58 @@
+    @override
+    def process(self, context: InitialAWSContext) -> None:
+        try:
+            context.aws_client = AWSClient(self._config, context.pm_account_id, self._role_name)
+        except AWSError as error:
+            raise UnexpectedStopError(
+                f"Program Management Account {context.pm_account_id} failed to retrieve "
+                f"credentials",
+                f"The Program Management Account {context.pm_account_id} is failing "
+                f"to provide valid AWS credentials with error: {error!s}. Please verify that "
+                f"the account ID is correct",
+            )
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)